source/templates.tex

```diff
@@ -2931,6 +2900,32 @@
 \tcode{,}       & \tcode{void()} \\
 \end{floattable}
 
+\pnum
+The instantiation of any other pack expansion
+produces a list of elements
+$\mathtt{E}_1,$ $\mathtt{E}_2,$ $\cdots,$ $\mathtt{E}_N$.
+\begin{note}
+The variety of list varies with the context:
+\grammarterm{expression-list},
+\grammarterm{base-specifier-list},
+\grammarterm{template-argument-list}, etc.
+\end{note}
+When $N$ is zero, the instantiation of the expansion produces an empty list.
+Such an instantiation does not alter the syntactic interpretation of the
+enclosing construct, even in cases where omitting the list entirely would
+otherwise be ill-formed or would result in an ambiguity in the grammar.
```

Comment:
Do we need to say the same thing for the fold-expression case? Eg:

```
int y;
template<int ...N> int x(int((y && ... && N)));
```

... is not interpreted as a function declaration `int x(int ((y)))` when the pack is empty. I think that's pre-existing, but if we're generalizing the E_i wording to apply to fold-expressions, then maybe we should consider generalizing this too? The intent seems clear that this should apply in all cases.

Reply:
Fixed.

Reply:
(@jensmaurer: can you please "resolve" conversations that definitely don't need further attention?)

Reply:
ack